lib/relaton_nist/hit_collection.rb

```diff
@@ -209,30 +142,49 @@ def from_ga # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
     #
     # @return [Array<RelatonNist::Hit>] hits
     #
-    def from_json
+    def from_json # rubocop:disable Metrics/AbcSize
+      return [] unless @pubid_ref.serie.serie.match?(/^(NIST\s(SP|CSWP)|FIPS)/)
+
       select_data.map do |h|
         /(?<series>(?<=-)\w+$)/ =~ h["series"]
         title = [h["title-main"], h["title-sub"]].compact.join " - "
         release_date = RelatonBib.parse_date h["published-date"], false
-        Hit.new({ code: docidentifier(h), series: series.upcase, title: title,
-                  url: h["uri"], status: h["status"],
-                  release_date: release_date, json: h }, self)
+        Hit.new({ code: json_pubid(h), series: series.upcase, title: title, url: h["uri"],
+                  status: h["status"], release_date: release_date, json: h }, self)
+      end
+    end
+
+    def json_pubid(json)
+      if json["doi"]
+        id = json["doi"].split("/").last.sub(/-draft\d*/, "").sub(/\.\wpd/, "").sub(/-(Add\d*)/, "")
+        add = Regexp.last_match(1)
+        id.sub!(/-upd(\d)/, "")
+        upd = Regexp.last_match(1)
+        pubid = Pubid::Nist::Identifier.parse id
+        pubid.addendum = add if add
+      else
+        id = json["docidentifier"]
+        id = "NIST #{id}" unless id.start_with? "FIPS"
+        upd = json["uri"]&.match(/\/upd(\d)\//)&.captures&.first
+        pubid = Pubid::Nist::Identifier.parse id
+        pubid.revision ||= json["revision"] if json["revision"]
+        pubid.edition ||= json["edition"] if json["edition"]
+        pubid.volume ||= json["volume"] if json["volume"]
       end
+      pubid.stage = create_stage json["iteration"]
+      pubid.update = Pubid::Nist::Update.new number: upd if defined?(upd) && upd
+      pubid
     end
 
-    def docidentifier(json) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
-      parts = doi_parts json
-      return json["docidentifier"] unless parts
+    def create_stage(iteration)
+      return unless iteration
 
-      id = parts[:code]
-      id = "NIST #{parts[:series]} #{id}" if parts[:series]
-      id += " Part #{parts[:prt]}" if parts[:prt]
-      id += " Vol. #{parts[:vol]}" if parts[:vol]
-      id += " Ver. #{parts[:ver]}" if parts[:ver]
-      id += " Rev. #{parts[:rev]}" if parts[:rev]
-      id += "-Add" if parts[:add]
-      id += " (Draft)" if parts[:draft] || @opts[:stage]
-      id
+      id =  case iteration
+            when "ipd" then "i"
+            when "final" then "f"
+            else iteration.match(/^\d+/)&.to_s
+            end
```

Comment:
@andrew2net can't we move all of these functions into pubid-nist?

Reply:
@ronaldtse I've created issues but they have not been solved yet. These are just temp solutions.